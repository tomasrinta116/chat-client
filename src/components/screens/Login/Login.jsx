import React from "react";
import TextField from "@material-ui/core/TextField";
import { withStyles } from "@material-ui/core/styles";
import LoginButton from "./LoginButton/LoginButton";
import styles from './Login.styles';
import PropTypes from 'prop-types';

class Login extends React.Component {
  state = {
    username: ""
  };

  handleChange = event => {
    this.setState({ username: event.target.value });
  };

  render() {
    const {
      classes: { loginContainer },
      history
    } = this.props;
    return (
      <div className={loginContainer}>
        <h1>react.chat</h1>
        <TextField
          id="username"
          placeholder="Enter a username"
          margin="normal"
          label="Username"
          InputLabelProps={{ shrink: true }}
          value={this.state.username}
          onChange={this.handleChange}
        />
        <LoginButton
          id="loginButton"
          setSB={this.props.setSB}
          history={history}
          username={this.state.username}
        />
      </div>
    );
  }
}

Login.propTypes = {
  history: PropTypes.object.isRequired,
  setSB: PropTypes.func.isRequired,
}

export default withStyles(styles)(Login);
