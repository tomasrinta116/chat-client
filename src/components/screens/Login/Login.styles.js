const styles = {
  /* --- <Login /> --- */
  loginContainer: {
    height: "100vh",
    display: "flex",
    flexDirection: "column",
    justifyContent: "center",
    alignItems: "center"
  }
};

export default styles;