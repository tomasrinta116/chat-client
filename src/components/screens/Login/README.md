# Login Overview
This folder contains all the contents for the Login screen and functionality. When a user enters a name and clicks the LoginButton, a SendBird object is retrieved and assigned to that user's individual instance of the app. The SendBird object contains the data and functions required for much of this application's core logic. The SendBird object is referred to as "sb" throughout the app codebase.