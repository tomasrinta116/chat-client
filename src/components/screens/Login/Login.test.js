import React from "react";
import {shallow} from 'enzyme';
import Login from "./Login";


describe("<Login />", () => {

    const props = {
        history: {},
        setSB: jest.fn(),
    }

    let wrapper; 
    beforeEach(() => {
        wrapper = shallow(<Login {...props} />).dive();
    })

    test("it renders", () => {
        const usernameTextField = wrapper.find({id: "username"});
        const loginButton = wrapper.find({id: "loginButton"});
        expect(usernameTextField).toHaveLength(1);
        expect(loginButton).toHaveLength(1);
    })

    test("the entire login page renders correctly", () => {
        expect(wrapper).toMatchSnapshot();
    })

    test("updating username is reflected in state", () => {
        const usernameTextField = wrapper.find({id: "username"});
        usernameTextField.simulate("change", { target: { value: 'test123'}});
        expect(wrapper.state().username).toEqual('test123');       
    })

})