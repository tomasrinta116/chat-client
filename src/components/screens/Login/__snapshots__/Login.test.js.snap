// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> the entire login page renders correctly 1`] = `
<div
  className="Login-loginContainer-1"
>
  <h1>
    react.chat
  </h1>
  <TextField
    InputLabelProps={
      Object {
        "shrink": true,
      }
    }
    id="username"
    label="Username"
    margin="normal"
    onChange={[Function]}
    placeholder="Enter a username"
    required={false}
    select={false}
    value=""
    variant="standard"
  />
  <LoginButton
    history={Object {}}
    id="loginButton"
    setSB={[MockFunction]}
    username=""
  />
</div>
`;
